File: smart_farm/src/api/crop.ts
import { get, http, post } from '@/utils/request'
import type { PageInfo } from '@/api/user'

/**
 * 作物信息接口封装，对应 smart_plant 后端 /crop 系列接口。
 */
export interface Crop {
  id?: number
  userId?: number
  username?: string
  nickname?: string
  typeId?: number | null
  typeName?: string
  cropName: string
  cropCode?: string
  variety?: string
  growthDays?: number
  suitableTemperature?: string
  suitableHumidity?: string
  suitablePh?: string
  imageUrl?: string
  description?: string
  status?: number
  remark?: string
  createTime?: string
  updateTime?: string
}

export interface CropPayload {
  typeId?: number | null
  cropName: string
  variety?: string
  growthDays?: number
  suitableTemperature?: string
  suitableHumidity?: string
  suitablePh?: string
  imageUrl?: string
  description?: string
  status?: number
  remark?: string
}

export interface UpdateCropPayload extends Partial<CropPayload> {
  id: number
}

export interface CropListParams {
  cropName?: string
  typeId?: number
  status?: number
  pageNum: number
  pageSize: number
}

interface ApiResult<T> {
  code: number
  message: string
  data: T
}

/** 分页查询作物信息列表。 */
export const listCrops = (params: CropListParams) => {
  return get<ApiResult<PageInfo<Crop>>>('/crop/list', params as unknown as Record<string, unknown>)
}

/** 根据 ID 查询作物详情。 */
export const getCropById = (id: number) => {
  return get<ApiResult<Crop>>(`/crop/${id}`)
}

/** 新增作物信息。 */
export const addCrop = (data: CropPayload) => {
  return post<ApiResult<Crop>>('/crop/add', data as unknown as Record<string, unknown>)
}

/** 编辑作物信息。 */
export const updateCrop = (data: UpdateCropPayload) => {
  return http<ApiResult<Crop>>({ url: '/crop', method: 'put', data })
}

/** 删除单个作物。 */
export const deleteCrop = (id: number) => {
  return http<ApiResult<void>>({ url: `/crop/${id}`, method: 'delete' })
}

/** 批量删除作物。 */
export const batchDeleteCrops = (ids: number[]) => {
  return http<ApiResult<number>>({ url: '/crop/batch', method: 'delete', data: ids })
}

/** 启用或禁用作物。 */
export const updateCropStatus = (id: number, status: number) => {
  return http<ApiResult<void>>({
    url: `/crop/${id}/status`,
    method: 'put',
    params: { status },
  })
}

File: smart_plant/src/main/java/com/smart_plant/smart_plant/entity/Crop.java
package com.smart_plant.smart_plant.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 作物信息实体，对应 crop 表。
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Crop {

    /** 作物ID */
    private Long id;

    /** 所属用户ID，空值表示系统公共作物 */
    private Long userId;

    /** 所属用户名，列表展示字段 */
    private String username;

    /** 所属用户昵称，列表展示字段 */
    private String nickname;

    /** 作物类型ID */
    private Long typeId;

    /** 作物类型名称，列表展示字段 */
    private String typeName;

    /** 作物名称 */
    private String cropName;

    /** 系统自动生成的作物编码 */
    private String cropCode;

    /** 品种名称 */
    private String variety;

    /** 推荐生长周期，单位天 */
    private Integer growthDays;

    /** 适宜温度范围 */
    private String suitableTemperature;

    /** 适宜湿度范围 */
    private String suitableHumidity;

    /** 适宜PH范围 */
    private String suitablePh;

    /** 适宜CO2浓度范围，如400-1000ppm */
    private String suitableCo2;

    /** 适宜PM2.5范围，如0-75μg/m³ */
    private String suitablePm25;

    /** 适宜EC值范围，如1.0-2.5mS/cm */
    private String suitableEc;

    /** 适宜溶解氧范围，如5.0-8.0mg/L */
    private String suitableDissolvedOxygen;

    /** 适宜光照强度范围，如10000-30000lux */
    private String suitableLight;

    /** 作物图片 */
    private String imageUrl;

    /** 作物说明 */
    private String description;

    /** 状态：1启用 0禁用 */
    private Integer status;

    /** 备注 */
    private String remark;

    /** 创建时间 */
    private LocalDateTime createTime;

    /** 更新时间 */
    private LocalDateTime updateTime;
}
